packages/@functionless/formation/src/resource-providers/cloud-control.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  CreateRequest,
+  DeleteRequest,
+  ResourceProvider as ResourceProvider,
+  ResourceProviderProps as ResourceProviderProps,
+  ResourceProviderRetryConfig,
+  UpdateRequest,
+} from "../resource-provider";
+import * as control from "@aws-sdk/client-cloudcontrol";
+import {
+  PhysicalProperties,
+  PhysicalResource,
+  ResourceType,
+} from "../resource";
+import { awsSDKRetry } from "../util";
+import { compare } from "fast-json-patch";
+
+/**
+ * Cloud control supports many, but not a complete set of all AWS resource.
+ *
+ * CC is also slower.
+ *
+ * https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/supported-resources.html
+ */
+export class CloudControlProvider
+  implements ResourceProvider<PhysicalProperties>
+{
+  readonly Type = "AWS::SQS::QueuePolicy";
+  private controlClient: control.CloudControlClient;
+
+  constructor(props: ResourceProviderProps) {
+    this.controlClient = new control.CloudControlClient(props.sdkConfig);
+  }
+  retry: ResourceProviderRetryConfig = { canRetry: true };
```

Comment:
Set this in the constructor? Make it read only?

Reply:
Moved it to the top and made it readonly. No reason to set in the constructor if it is a constant.

Reply:
Why even have it if it's a constant? I was just thinking for consistency. It's jarring code

Reply:
Its configuration per provider. Not all providers can idempotently be retried on all operations.